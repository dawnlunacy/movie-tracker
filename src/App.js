import React from 'react';
import { BrowserRouter as Router, Switch, Route} from 'react-router-dom'
import logo from './logo.svg';
import './App.css';


function App() {
  return (
    <Router>
    <div className="App">
      <header className="App-header">
        <img src={logo} className="App-logo" alt="logo" />
        <p>
          HIIIII
          Edit <code>src/App.js</code> and save to reload.
        </p>
        <a
          className="App-link"
          href="https://reactjs.org"
          target="_blank"
          rel="noopener noreferrer"
        >
          Learn React
        </a>
      </header>
    </div>
    </Router>
  );
}

export default App;
