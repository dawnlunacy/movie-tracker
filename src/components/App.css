.App {
  display: flex;
  text-align: center;
}

.App-logo {
  height: 40vmin;
}

.App-header {
  display: flex;
  align-items: center;
  justify-content: center;
  width: 100%;
}

.App-link {
  color: #09d3ac;
}

.App-img {
  margin: 25px;
  position: absolute;
  top: 30px;
  width: 300px;
  height: 40vmin;
}
