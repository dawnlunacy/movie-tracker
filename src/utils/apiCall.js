import { validate } from "@babel/types";

export const fetchData = async(url)  => {
  const response = await fetch(url)
  if(response.ok) {
    const data  = await response.json()
    return data;
  } else {
    throw Error(response.statusText)
  }
}

export const loginUser = async (userInfo, url) => {
  const options = {
    method: 'POST',
    body: JSON.stringify(userInfo),
    headers: {
      'Content-Type': 'application/json'
    }
  }

  const response = await fetch(url, options)
    if (!response.ok) {
      throw Error(response.statusText)
    } 
    const validatedUser = await response.json()
    console.log("VUSER", validatedUser)
    return validatedUser
    
}
