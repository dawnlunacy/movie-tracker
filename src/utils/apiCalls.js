
export const fetchData = async(url)  => {
  const response = await fetch(url)
  if(response.ok) {
    const data  = await response.json()
    return data;
  } else {
    throw Error(response.statusText)
  }
}

export const getUser = async (userInfo, url) => {
  const options = {
    method: 'POST',
    body: JSON.stringify(userInfo),
    headers: {
      'Content-Type': 'application/json'
    }
  }

  const response = await fetch(url, options)
    return response
}

export const postFavorite = async (movieInfo, id) => {
  const options = {
    method: 'POST',
    body: JSON.stringify(movieInfo),
    headers: {
      'Content-Type': 'application/json'
    }
  }

  const response = await fetch(`http://localhost:3001/api/v1/users/${id}/moviefavorites`, options)
  const data = await response.json();
  console.log('data', data)
    return data
}
