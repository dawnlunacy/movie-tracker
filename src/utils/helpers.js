export const filteredMovieData = movies => {
  return movies.map(movie => ({
    id: movie.id,
    title: movie.title,
    poster: movie.poster_path,
    rating: movie.vote_average,
    overview: movie.overview
  }))
}