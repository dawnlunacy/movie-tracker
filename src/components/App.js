import React, { Component } from 'react';
import { BrowserRouter as Router, Switch, Route} from 'react-router-dom';
import MoviesContainer from '../containers/MoviesContainer/MoviesContainer';
import Nav from '../containers/Nav/Nav';
import UserForm from '../containers/UserForm/UserForm';
import { connect } from 'react-redux';
import { bindActionCreators } from 'redux';
import { getMovies } from '../actions';
import { fetchData } from '../utils/apiCall';
import { filteredMovieData } from '../utils/helpers';
import './App.css';


class App extends Component {

  componentDidMount = async () =>  {
    const { getMovies } = this.props
    console.log('in cpm--->', this.props)
    const movies = await fetchData('https://api.themoviedb.org/3/movie/now_playing?api_key=cd7eb6a4cff8273d777385057dcf9b56')
    const cleanMovies = filteredMovieData(movies.results)
    getMovies(cleanMovies)
  }

  render() {
    const { movies } = this.props
    return (
      <Router>
        <div>
          <h1>Movie Tracker</h1>
          <header>
            <Nav />
          </header>
          <MoviesContainer movies={movies}/>
          <UserForm />
        </div>
      </Router>
    );
  }
}

export const mapStateToProps = state => ({
  movies: state.movies
});

export const mapDispatchToProps = dispatch => (
  bindActionCreators(
    {
      getMovies
    },
  dispatch)
)

export default connect(mapStateToProps, mapDispatchToProps)(App);
