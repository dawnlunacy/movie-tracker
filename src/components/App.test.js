import React from 'react';
import { shallow } from 'enzyme';
import { App, mapStateToProps, mapDispatchToProps } from './App';
import { fetchData } from '../utils/apiCalls';

jest.mock('../utils/apiCalls')

describe('App', () => {
  let wrapper;
  const mockGetMovies = jest.fn();
  const mockhandleError = jest.fn();
  const mockIsLoading = jest.fn();
  const mockSaveUser = jest.fn();
  const mockCleanMovies = jest.fn();

  const mockCurrentUser = {
    id: 3, 
    email:'rudd.lacy@gmail.com', 
    name: 'lacy'
  };

  fetchData.mockImplementation(() => Promise.resolve(mockCurrentUser))

  beforeEach(() => {
    wrapper = shallow(<App 
      currentUser={ mockCurrentUser }
      loading = {false}
      getMovies = {mockGetMovies}
      handleError = {mockhandleError}
      isLoading = {mockIsLoading}
      saveUser = {mockSaveUser}
      cleanMovies = {mockCleanMovies}
      />);
  });

  it('should match snapshot', () => {
    expect(wrapper).toMatchSnapshot()
  });
  
  
  describe('getFavorites', () => {
    let mockResponse = {
        "results": [
          {
            "poster_path": "/udDclJoHjfjb8Ekgsd4FDteOkCU.jpg",
            "id": 475557,
            "original_title": "Joker",
              "title": "Joker",
              "vote_average": 8.6,
              "overview": "During the 1980s, a failed stand-up comedian is driven insane and turns to a life of crime and chaos in Gotham City while becoming an infamous psychopathic crime figure.",
              "release_date": "2019-10-04"
            },
          ]};
          
          beforeEach(() => {
            window.fetch = jest.fn().mockImplementation(() => {
              return Promise.resolve({
                ok: true,
                json: () => Promise.resolve(mockResponse)
              });
            });
      });
      
      it ('should return array of favorited movies for currentUser', async () => {
        const mockUrl = `http://localhost:3001/api/v1/users/${mockCurrentUser.id}/moviefavorites`;
        
        await wrapper.instance().getFavorites(mockCurrentUser.id);
        
        fetchData(mockUrl)
        .then(results => expect(results).toEqual(mockResponse.results))
        .catch(error => error)
      });
    });

    describe('mapStateToProps', () => {
      
      let mockCurrentUser;
      beforeEach(() => {
        mockCurrentUser = {
          id: 3, 
          email:'rudd.lacy@gmail.com', 
          name: 'lacy'
        };

      })
      it('should return an object with the currentUser, isLoading, and favorited', () => {
        const mockState = { 
          currentUser: {currentUser: mockCurrentUser},
          loading: false,
          favorited: [           
            {
              "poster_path": "/udDclJoHjfjb8Ekgsd4FDteOkCU.jpg",
              "id": 475557,
              "original_title": "Joker",
              "title": "Joker",
              "vote_average": 8.6,
              "overview": "During the 1980s, a failed stand-up comedian is driven insane and turns to a life of crime and chaos in Gotham City while becoming an infamous psychopathic crime figure.",
              "release_date": "2019-10-04"
            }]
        };
        const expected = {
          currentUser: {currentUser: mockCurrentUser},
          loading: false,
          favorited: mockState.favorited
        };
    
        const mappedProps = mapStateToProps(mockState);
    
        expect(mappedProps).toEqual(expected);
      });

    });

    describe.skip('mapDispatchToProps', () => {
      it('calls dispatch with a getUser action is called', () => {
        const mockDispatch = jest.fn();
        const movies =           
        {
          "poster_path": "/udDclJoHjfjb8Ekgsd4FDteOkCU.jpg",
          "id": 475557,
          "original_title": "Joker",
          "title": "Joker",
          "vote_average": 8.6,
          "overview": "During the 1980s, a failed stand-up comedian is driven insane and turns to a life of crime and chaos in Gotham City while becoming an infamous psychopathic crime figure.",
          "release_date": "2019-10-04"
          }
  
        const actionToDispatch = wrapper.instance().getMovies(movies);
  
        const mappedProps = mapDispatchToProps(mockDispatch);
        mappedProps.wrapper.instance().getMovies(movies);
  
        expect(mockDispatch).toHaveBeenCalledWith(actionToDispatch);
      });

    });
    
  }); 
