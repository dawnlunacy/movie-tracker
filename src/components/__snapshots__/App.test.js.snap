// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App should match snapshot 1`] = `
<div
  className="App"
>
  <Route
    exact={true}
    path="/login"
    render={[Function]}
  />
  <Route
    exact={true}
    path="/signup"
    render={[Function]}
  />
  <Route
    exact={true}
    path="/"
    render={[Function]}
  />
  <Route
    exact={true}
    path="/movies/:id"
    render={[Function]}
  />
  <Route
    exact={true}
    path="/favorites"
    render={[Function]}
  />
</div>
`;
